### DISCUSSION TITLE
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL 

### DISCUSSION THREAD

**Discussion by anlexN** [directus/directus#15392] [Archive]
### Describe the Bug

I am reviewing directus source code. Build is failed
![image](https://user-images.githubusercontent.com/8686739/188307610-eabaa948-725b-4fb9-b178-afe06eb1e8d3.png)
![image](https://user-images.githubusercontent.com/8686739/188307631-3008b4ac-a684-4ce6-abc0-e5ca3ad11e01.png)

### To Reproduce

1. git clone git@github.com:YOUR-USERNAME/directus.git
2. git checkout -b YOUR-BRANCH-NAME
3. pnpm install (success)
4. pnpm -r build (failed)

### Errors Shown

 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @directus/app@9.17.0 build: `vite build`
Exit status 134

### What version of Directus are you using?

9.14.2

### What version of Node.js are you using?

16.17.0

### What database are you using?

Postgres 14

### What browser are you using?

all

### How are you deploying Directus?

locally

**Comment by azrikahar** [comment]
Seems similar to https://github.com/directus/directus/discussions/14348#discussioncomment-3112522. You can try setting the `--max_old_space_size` flag as described in the linked comment to get around the memory restriction in your machine 👍

**Comment by Autr** [comment]
Hi, I had the same problem on M1 Mac running dev - tried `NODE_OPTIONS=--max_old_space_size=8192 pnpm -r dev`  etc.

What worked was just running app & api in two terminals like in the guide:

* API `pnpm --filter directus dev`
* App `pnpm --filter @directus/app dev`

Some other set up issues I had were with sharp and libvips on M1. This helped, if not the most elegant! 🫣

```zsh
brew uninstall vips sqlite3
brew install sqlite3
pnpm i --save sqlite3
SHARP_IGNORE_GLOBAL_LIBVIPS=true pnpm i
``
